hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py

```diff
@@ -90,46 +95,69 @@ def lambda_handler(event, context):
-    action_labels: t.List["ActionLabel"] = []
-    for action_rule in action_rules:
-        if action_rule_applies_to_match_message(
-            action_rule, match_message
-        ) and not action_labels.__contains__(action_rule.action_label):
-            action_labels.append(action_rule.action_label)
+    classifications_by_match = get_classifications_by_match(match_message)
+    action_labels: t.Set[ActionLabel] = set()
+    for classifications in classifications_by_match:
+        for action_rule in action_rules:
+            if action_rule_applies_to_classifications(action_rule, classifications):
+                action_labels.add(action_rule.action_label)
     action_labels = remove_superseded_actions(action_labels)
     return action_labels
 
 
-def get_action_rules() -> t.List["ActionRule"]:
+def get_classifications_by_match(match_message: MatchMessage) -> t.List[t.Set[Label]]:
     """
-    TODO implement
+    Creates a list of sets of classifications (as labels). Each set contains the labels that
+    classify one matching banked piece of content.
+    """
+    classifications_by_match: t.List[t.Set[Label]] = list()
+
+    for banked_signal in match_message.matching_banked_signals:
+        classifications: t.Set[Label] = set()
+        classifications.add(Label("BankSource", banked_signal.bank_source))
+        classifications.add(Label("BankId", banked_signal.bank_id))
+        classifications.add(Label("BankedContentId", banked_signal.banked_content_id))
+        for classification in banked_signal.classifications:
+            classifications.add(ClassificationLabel(classification))
+        classifications_by_match.append(classifications)
+
+    return classifications_by_match
+
+
+def get_action_rules() -> t.List[ActionRule]:
+    """
+    TODO implement (get from config)
     Returns the ActionRule objects stored in the config repository. Each ActionRule
     will have the following attributes: MustHaveLabels, MustNotHaveLabels, ActionLabel.
     """
     return [
         ActionRule(
             ActionLabel("EnqueueForReview"),
-            [Label("Collaboration", "12345")],
-            [],
+            [Label("BankId", "12345")],
+            [Label("Classification", "Foo")],
         )
     ]
 
 
-def action_rule_applies_to_match_message(
-    action_rule: ActionRule, match_message: MatchMessage
+def action_rule_applies_to_classifications(
+    action_rule: ActionRule, classifications: t.Set[Label]
 ) -> bool:
     """
-    Evaluate if the action rule applies to the match message. Return True if the action rule's "must have"
-    labels are all present in the match message, and that none of the "must not have" labels are present
-    in the match message, otherwise return False.
+    Evaluate if the action rule applies to the classifications. Return True if the action rule's "must have"
+    labels are all present and none of the "must not have" labels are present in the classifications, otherwise return False.
```

Comment:
Next PR is to implement action rule config, at which time I'll make `must_have_labels` and `must_not_have_labels` sets so they don't have to be converted from `List` to `Set` here.